lib/blockchain/chain.js

```diff
@@ -100,8 +100,12 @@ class Chain extends AsyncEmitter {
 
     this.setDeploymentState(state);
 
-    if (!this.options.spv)
-      await this.syncTree();
+    if (!this.options.spv) {
+      if (this.options.compactTree)
+        await this.compactTree();
```

Comment:
OK, I like this too. Had to modify tests to manually call `syncTree()` in some places but the separation makes sense. And since we removed the rpc call especially since this only happens in `open()` now.